crypto/txscript/src/lib.rs

```diff
@@ -466,12 +524,9 @@ impl<'a, T: VerifiableTransaction, Reused: SigHashReusedValues> TxScriptEngine<'
         } else if num_sigs > num_keys {
             return Err(TxScriptError::InvalidSignatureCount(format!("more signatures than pubkeys {num_sigs} > {num_keys}")));
         }
-        if let Some(RuntimeSigOpCounter { sig_op_remaining, sig_op_limit }) = &mut self.runtime_sig_op_counting {
-            let num_sigs =
-                num_sigs.try_into().map_err(|_| TxScriptError::NumberTooBig("Signatures count mustn't exceed 255".to_string()))?;
-            *sig_op_remaining =
-                sig_op_remaining.checked_sub(num_sigs).ok_or(TxScriptError::ExceededSigOpLimit(num_sigs as u16, *sig_op_limit))?;
-        }
+        let num_sigs =
+            num_sigs.try_into().map_err(|_| TxScriptError::NumberTooBig("Signatures count mustn't exceed 255".to_string()))?;
```

Comment:
I had a suspicion that this can create a netsplit, since `num_sigs` arrived from the stack as `i32` so it can theoretically be bigger than 255. But after some reading of the context it seems that this error is actually unreachable, since `num_sigs` is bounded by `num_keys` which is bounded by `MAX_PUB_KEYS_PER_MUTLTISIG=20`.

But just to be sure, can you make a multisig test where the hardfork is inactive and `num_sigs=256`? I expect to see it fails with `Err(TxScriptError::InvalidPubKeyCount(format!("too many pubkeys {num_keys} > {MAX_PUB_KEYS_PER_MUTLTISIG}")))` or `Err(TxScriptError::InvalidSignatureCount(format!("more signatures than pubkeys {num_sigs} > {num_keys}")))`

Reply:
actually it will face first  MAX_SCRIPT_ELEMENT_SIZE then MAX_STACK_SIZE and only after that, it will fail with error related to MAX_PUB_KEYS_PER_MUTLTISIG.

Reply:
7fedccaedf601971b284e7d0659576961835876f